src/stability_sdk/animation.py

```diff
@@ -163,65 +151,75 @@ def args2dict_simplenamespace(args):
+        self.start_frame_idx: int = 0
+        self.video_prev_frame: np.ndarray = None
+        self.video_reader = None
+
+        self.stub = stub
         self.transform_engine_id = transform_engine_id
         self.inpaint_engine_id = inpaint_engine_id
         self.generate_engine_id = generate_engine_id
 
-        self.video_prev_frame = None
-        self.setup_animation()
+        self.setup_animation(resume)
 
-    def get_animation_prompts_weights(
-        self,
-        frame_idx: int,
-        key_frame_values: List[int],
-        interp: bool
-    ) -> Tuple[List[str], List[float]]:
-        idx = bisect.bisect_right(key_frame_values, frame_idx)
+    def get_animation_prompts_weights(self, frame_idx: int) -> Tuple[List[str], List[float]]:
+        keys = self.key_frame_values
+        idx = bisect.bisect_right(keys, frame_idx)
         prev, next = idx - 1, idx
-        if not interp:
-            return [self.animation_prompts[key_frame_values[min(len(key_frame_values)-1, prev)]]], [1.0]
-        elif next == len(key_frame_values):
-            return [self.animation_prompts[key_frame_values[-1]]], [1.0]
+        if not self.args.interpolate_prompts:
+            return [self.animation_prompts[keys[min(len(keys)-1, prev)]]], [1.0]
+        elif next == len(keys):
+            return [self.animation_prompts[keys[-1]]], [1.0]
         else:
-            tween = (frame_idx - key_frame_values[prev]) / (key_frame_values[next] - key_frame_values[prev])
-            return [self.animation_prompts[key_frame_values[prev]], self.animation_prompts[key_frame_values[next]]], [1.0 - tween, tween]
-
-
-    def save_settings(self):
-            # save settings for the animation
-            timestring = datetime.datetime.now().strftime('%Y%m%d%H%M%S')
-            settings_filename = os.path.join(self.out_dir, f"{timestring}_settings.txt")
-            with open(settings_filename, "w+", encoding="utf-8") as f:
-                save_dict = args2dict(self.args)
-                for k in ['angle', 'zoom', 'translation_x', 'translation_y', 'translation_z', 'rotation_x', 'rotation_y', 'rotation_z']:
-                    save_dict.move_to_end(k, last=True)
-                save_dict['animation_prompts'] = self.animation_prompts
-                save_dict['negative_prompt'] = self.negative_prompt
-                save_dict['negative_prompt_weight'] = self.negative_prompt_weight
-                json.dump(save_dict, f, ensure_ascii=False, indent=4)
-
-    def setup_animation(self):
+            tween = (frame_idx - keys[prev]) / (keys[next] - keys[prev])
+            return [self.animation_prompts[keys[prev]], self.animation_prompts[keys[next]]], [1.0 - tween, tween]
+
+    def get_frame_filename(self, frame_idx, depth=False):
+        prefix = "depth" if depth else "frame"
+        return os.path.join(self.out_dir, f"{prefix}_{frame_idx:05d}.png")
+
+    def save_settings(self, filename: str):
```

Comment:
this whole process could be simplified if we migrate config and argument management to omegaconf (or something similar - I personally like oc but I'm open to considering gin or whatever)

Reply:
it's got a little extra complication due to request of having the prompts and camera settings at bottom so testers can easily remove those before sharing their settings. not opposed to any other ways of saving settings. they're primarily for users to go back and find settings they used in past runs. could be yaml, ini, or whatever. json atm to match what people are used to.